Notify/Notify/Notify/Services/Location/LocationServices.cs

```diff
@@ -0,0 +1,152 @@
+                    {
+                        Device.BeginInvokeOnMainThread(() =>
+                        {
+                            var errormessage = new LocationErrorMessage();
+                            MessagingCenter.Send(errormessage, "LocationError");
+                        });
+                    }
+                }
+            }, token);
+        }
+
+        public void Current_PositionChanged(object sender, Plugin.Geolocator.Abstractions.PositionEventArgs e)
+        {
+            Console.WriteLine($"{e.Position.Latitude}, {e.Position.Longitude}, {e.Position.Timestamp.TimeOfDay}");
+        }
+
+        public async Task ManageLocationTracking()
+        {
+            var permission = await Permissions.RequestAsync<Permissions.LocationAlways>();
+
+            if (permission == PermissionStatus.Denied)
+            {
+                // TODO Let the user know they need to accept
+                return;
+            }
+
+            if (Device.RuntimePlatform == Device.iOS)
+            {
+                if (CrossGeolocator.Current.IsListening)
+                {
+                    await CrossGeolocator.Current.StopListeningAsync();
+                    CrossGeolocator.Current.PositionChanged -= Current_PositionChanged;
+
+                    return;
+                }
+
+                await CrossGeolocator.Current.StartListeningAsync(TimeSpan.FromSeconds(1), 10, false,
+                    new Plugin.Geolocator.Abstractions.ListenerSettings
+                    {
+                        ActivityType = Plugin.Geolocator.Abstractions.ActivityType.AutomotiveNavigation,
+                        AllowBackgroundUpdates = true,
+                        DeferLocationUpdates = false,
+                        DeferralDistanceMeters = 10,
+                        DeferralTime = TimeSpan.FromSeconds(5),
+                        ListenForSignificantChanges = true,
+                        PauseLocationUpdatesAutomatically = true
+                    });
+
+                CrossGeolocator.Current.PositionChanged += Current_PositionChanged;
+            }
+            else if (Device.RuntimePlatform == Device.Android)
+            {
+                if (Preferences.Get("LocationServiceRunning", false) == false)
+                {
+                    StartService();
+                }
+                else
+                {
+                    StopService();
+                }
+            }
+        }
+        
+        public void SubscribeToLocationMessaging()
+        {
+            if (Device.RuntimePlatform == Device.Android)
+            {
+                MessagingCenter.Subscribe<LocationMessage>(this, "Location",
+                    message =>
+                    {
+                        Device.BeginInvokeOnMainThread(() =>
+                        {
+                            Console.WriteLine(
+                                $"{message.Latitude}, {message.Longitude}, {DateTime.Now.ToLongTimeString()}");
```

Comment:
I think we should add "User's current location:" in the beginning of the message, and to print it to console only when we're on debug mode.

Reply:
Done